File: 3-leetcode/02-medium/js/2109-adding-spaces-to-a-string.ts
const addSpaces = (s: string, spaces: number[]): string => {
  let result = '';

  let spaceIndex = 0;

  for (let stringIndex = 0; stringIndex < s.length; stringIndex++) {
    if (spaceIndex < spaces.length && stringIndex == spaces[spaceIndex]) {
      result += ' ';
      spaceIndex++;
    }

    result += s[stringIndex];
  }

  return result;
};

File: 3-leetcode/02-medium/dart/2109-adding-spaces-to-a-string.dart
class Solution {
  String addSpaces(String s, List<int> spaces) {
    StringBuffer result = StringBuffer();

    int spaceIndex = 0;

    for (int stringIndex = 0; stringIndex < s.length; stringIndex++) {
      if (spaceIndex < spaces.length && stringIndex == spaces[spaceIndex]) {
        result.write(' ');
        spaceIndex++;
      }

      result.write(s[stringIndex]);
    }

    return result.toString();
  }
}
